python/paddle/fluid/tests/unittests/distribute/dist_test.py

```diff
@@ -0,0 +1,129 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import subprocess
+import time
+import argparse
+import sys
+import os
+import signal
+from contextlib import closing
+import socket
+
+JOB_ROLE_MASTER = "MASTER"
+JOB_ROLE_TRAINER = "TRAINER"
+JOB_ROLE_PSERVERS = "PSERVER"
+
+# detect cuda device count automatically
+CUDA_DEVICE_CNT = 2
+
+
+class FluidDistTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.entry = ""
+        cls.place = None
+
+    def _wait_ps_ready(self, pid):
+        retry_times = 5
+        while True:
+            assert retry_times >= 0, "wait ps ready failed"
+            time.sleep(1)
+            try:
+                # the listen_and_serv_op would touch a file which contains the listen port
+                # on the /tmp directory until it was ready to process all the RPC call.
+                os.stat("/tmp/paddle.%d.port" % pid)
+                return
+            except os.error:
+                retry_times -= 1
+
+    def _find_free_port(self):
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+            s.bind(("localhost", 0))
+            return s.getsockname()[1]
+
+    def _ps_health(self, p):
+        if p is None and p != 0:
+            return False
+        else:
+            return True
+
+    def launch_job(self, trainers, pservers):
+        """
+        This function will launch a distributed training job and hung until
+        one of trainers or pservers exit.
+        """
+        role = os.getenv("PADDLE_TRAINING_ROLE", "MASTER")
+        ps_endpoints = ",".join([
+            "127.0.0.1:%d" % self._find_free_port() for _ in xrange(pservers)
+        ])
+
+        if role == JOB_ROLE_MASTER:
+            p_list = []
+
+            for i in xrange(pservers):
+                p_env = os.environ
+                p_env["PADDLE_TRAINING_ROLE"] = "PSERVER"
```

Comment:
envs like `PATH`, `LD_LIBRARY_PATH`, `PYTHONPATH` should be put into, when using manylinux image to run test these are needed.

Reply:
the sub-process will inherit the env from master process by `os.environ`.